service/status-report/main.go

```diff
@@ -132,6 +137,20 @@ func main() {
 		Name:      storageClient.Name,
 	}
 
+	clusterConfig := &v1.ConfigMap{
+		ObjectMeta: v12.ObjectMeta{Namespace: clusterConfigNamespace, Name: clusterConfigName},
+	}
+	if err = cl.Get(ctx, client.ObjectKeyFromObject(clusterConfig), clusterConfig); err != nil {
+		klog.Exitf("Failed to get clusterConfig %q/%q: %v", clusterConfig.Namespace, clusterConfig.Name, err)
+	}
+	clusterMetadata := clusterConfig.Data["metadata"]
+	var clusterMetadataRes map[string]string
+	err = json.Unmarshal([]byte(clusterMetadata), &clusterMetadataRes)
+	if err != nil {
+		klog.Exitf("Fatal error, %v", err)
+	}
+	clusterName := clusterMetadataRes["name"]
```

Comment:
I don't think this is correct, as per the sample config you posted earlier there's yaml value for `install-config` key stored in the configmap and below might work
``` go
        installConfig := clusterConfig.Data["install-config"]
	metadata := struct {
		Metadata struct {
			Name string `yaml:"name"`
		} `yaml:"metadata"`
	}{}
	yaml.Unmarshal([]byte(installConfig), &metadata)
        fmt.Println(metadata.Name) // check for emptiness as well if required
```